src/Listener.ts

```diff
@@ -98,6 +99,7 @@ export default class Listener extends EventEmitter {
   }
 
   destroy() {
+    clearTimeout(this.requeryDelay);
     this.server.removeListener("response", this.onResponse);
   }
```

Comment:
I think we are missing a destroy emit here.

```suggestion
    this.emit("destroy");
  }
```